Core/TextureReplacer.cpp

```diff
@@ -193,26 +132,26 @@ bool TextureReplacer::LoadIni() {
 		if (ini.GetOrCreateSection("games")->Get(gameID_.c_str(), &overrideFilename, "")) {
 			if (!overrideFilename.empty() && overrideFilename != INI_FILENAME) {
 				IniFile overrideIni;
-				if (zip_.z) {
-					std::lock_guard<std::mutex> guard(zip_.lock);
-					iniLoaded = LoadIniZip(overrideIni, zip_.z, overrideFilename);
-				} else {
-					iniLoaded = overrideIni.LoadFromVFS(g_VFS, (basePath_ / overrideFilename).ToString());
-				}
+				iniLoaded = overrideIni.LoadFromVFS(*dir, overrideFilename);
 				if (!iniLoaded) {
 					ERROR_LOG(G3D, "Failed to load extra texture ini: %s", overrideFilename.c_str());
+					// TODO: Should we just move on instead?
+					delete dir;
 					return false;
 				}
 
 				INFO_LOG(G3D, "Loading extra texture ini: %s", overrideFilename.c_str());
 				if (!LoadIniValues(overrideIni, true)) {
+					delete dir;
 					return false;
 				}
 			}
 		}
 	}
 
-	// The ini doesn't have to exist for it to be valid.
+	vfs_ = dir;
+
+	// The ini doesn't have to exist for the texture directory or zip to be valid.
```

Comment:
Well, I thought of this when we started supporting zips.  I left not requiring textures.ini for directories because I didn't want to break BC.  That said, I think most texture packs do use textures.ini in some way so it probably wouldn't be that bad.  I think it's a reasonable BC break...

-[Unknown]

Reply:
Oh, right, backwards compatibility..

No, I think I'm leaning towards printing a warning in this case instead (and disallowing ini-less texture zips).